lib/sync_checker/formats/edition_base.rb

```diff
@@ -26,6 +26,7 @@ def edition_expected_in_draft
       end
 
       def edition_expected_in_live
+        return nil if latest_edition_is_rejected?(document)
```

Comment:
I was more inclined to go the the other way and turn the other two cases into methods that describe them as if you're not familiar with all the vagaries of `Document` and `Edition` it isn't clear what these cases actually represent.